src/StorageCache.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace BlackBox;
+
+use IteratorAggregate;
+
+/**
+ * Caches an storage with another storage.
```

Comment:
Typo:

> Caches *a* storage with another storage.

Also could you add a line below saying that iterating this storage will not use the cache at all? At least that would make it explicit to users.